instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java

```diff
@@ -189,4 +205,164 @@ public void onResponse(Call call, Response response) {
+                    span.hasName("test-http-server")
+                        .hasKind(SpanKind.SERVER)
+                        .hasParent(trace.getSpan(0))),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttribute(
+                            SemanticAttributes.HTTP_URL, uri.resolve("/success").toString())
+                        .hasAttribute(SemanticAttributes.HTTP_RESEND_COUNT, 1L),
+                span ->
+                    span.hasName("test-http-server")
+                        .hasKind(SpanKind.SERVER)
+                        .hasParent(trace.getSpan(0))));
+  }
+
+  @Test
+  void new_basicRequestWith2Redirects() throws Exception {
+    URI uri = resolveAddress("/another-redirect");
+
+    int responseCode = sendRequest(buildRequest("GET", uri, emptyMap()), "GET", uri, emptyMap());
+
+    assertThat(responseCode).isEqualTo(200);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttribute(SemanticAttributes.HTTP_URL, uri.toString()),
+                span ->
+                    span.hasName("test-http-server")
+                        .hasKind(SpanKind.SERVER)
+                        .hasParent(trace.getSpan(0))),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttribute(
+                            SemanticAttributes.HTTP_URL, uri.resolve("/redirect").toString())
+                        .hasAttribute(SemanticAttributes.HTTP_RESEND_COUNT, 1L),
+                span ->
+                    span.hasName("test-http-server")
+                        .hasKind(SpanKind.SERVER)
+                        .hasParent(trace.getSpan(0))),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttribute(
+                            SemanticAttributes.HTTP_URL, uri.resolve("/success").toString())
+                        .hasAttribute(SemanticAttributes.HTTP_RESEND_COUNT, 2L),
+                span ->
+                    span.hasName("test-http-server")
+                        .hasKind(SpanKind.SERVER)
+                        .hasParent(trace.getSpan(0))));
+  }
+
+  static final String BASIC_AUTH_KEY = "custom-authorization-header";
+  static final String BASIC_AUTH_VAL = "plain text auth token";
```

Comment:
I think these can just be inlined -- I only see the one use. Or maybe they're used in children? 🤷🏻